packages/cli/src/index.ts

```diff
@@ -390,41 +390,32 @@ const main = async () => {
       const fileType = lstatSync(targetPath).isDirectory()
         ? 'subdirectory'
         : 'file';
-      const plural = targetOrSubcommand + 's';
-      const singular = targetOrSubcommand.endsWith('s')
-        ? targetOrSubcommand.slice(0, -1)
-        : '';
-      let alternative = '';
-      if (commands.has(plural)) {
-        alternative = plural;
-      } else if (commands.has(singular)) {
-        alternative = singular;
-      }
-      console.error(
-        error(
-          `The supplied argument ${param(targetOrSubcommand)} is ambiguous.` +
-            `\nIf you wish to deploy the ${fileType} ${param(
-              targetOrSubcommand
-            )}, first run "cd ${targetOrSubcommand}". ` +
-            (alternative
-              ? `\nIf you wish to use the subcommand ${param(
-                  targetOrSubcommand
-                )}, use ${param(alternative)} instead.`
-              : '')
-        )
+
+      output.warn(
+        `Did you mean to deploy the ${fileType} "${targetOrSubcommand}"? ` +
+          `Use \`vc --cwd ${targetOrSubcommand}\` instead.`
       );
-      return 1;
     }
 
     if (subcommandExists) {
       debug(`user supplied known subcommand: "${targetOrSubcommand}"`);
       subcommand = targetOrSubcommand;
     } else {
       debug('user supplied a possible target for deployment');
+
+      if (targetOrSubcommand !== '--cwd') {
+        output.warn(
+          'calling `vc` without a command or the `--cwd` option is deprecated, use `vc --cwd directory` or `vc deploy` instead'
+        );
+      }
+
       subcommand = 'deploy';
     }
   } else {
     debug('user supplied no target, defaulting to deploy');
+    output.warn(
+      'calling `vc` without a command or the `--cwd` option is deprecated, use `vc --cwd directory` or `vc deploy` instead'
+    );
```

Comment:
Do we want to warn in the general case like this since there is no conflict?

Seems like its one of the nice things to be able to type two letters to deploy: `vc`

Reply:
I agree that is nice. I think it would be fine to leave the `vc` invocation alone. I called it out in the description under "Alternatives".

@TooTallNate what do you think?

Reply:
`vc` by itself is fine but `vc <some-dir>` should print a warning, otherwise someone might rely on that in i.e. their CI and that makes it hard to add new subcommands (`vc build` to deploy the static "build" directory for example).

Reply:
I removed the deprecation warning

Reply:
Let's update the PR description as well since it says the bare `vc` version prints a warning which has now been changed.

Reply:
@EndangeredMassa The PR title and description are still outdated

Reply:
I think the title is still correct because it's talking about `vc <dir>`, not `vc`. I updated the description.

Reply:
How about `Add support for vc --cwd <dir>` since the warning doesn't say anything about deprecation

Reply:
That is better. Updated.